content/src/debug.js

```diff
@@ -206,39 +219,190 @@ function reportSeeeval(method, debugId, length, coordId, elem, args){
   debugRecordsByDebugId[currentDebugId] = record;
 }
 
+function reportEnter(method, debugId, length, coordId, elem, args){
+  var recordD = debugRecordsByDebugId[debugId];
+  var recordL = debugRecordsByLineNo[recordD.line];
+  recordD.animated = false;
+  recordL.animated = false;
+}
+
+
 function reportAppear(method, debugId, length, coordId, elem, args){
-  stuckComplexity.moves += 1;
+if (!programChanged) { 
   var recordD = debugRecordsByDebugId[debugId];
-  if (recordD) {
-    if (!recordD.seeeval) {
-      var recordL = debugRecordsByLineNo[recordD.line];
-      recordD.method = method;
-      recordL.method = method;
-      recordD.args = args;
-      recordL.args = args;
+  var recordL = debugRecordsByLineNo[recordD.line];
+  recordD.method = method;
+  recordL.method = method;
+  recordD.args = args;
+  recordL.args = args;
+
+  var currentRecord = debugRecordsByDebugId[currentRecordID];
+  var currentIndex = currentRecord.eventIndex;
+  var currentLocation = traceEvents[currentIndex].location;
+  var currentLine = currentLocation.first_line;
+
+  var prevLine = -1;
+
+  stuckComplexity.moves += 1;
+
+  if (recordD) { 
+    if (!recordD.seeeval){ 
       var index = recordD.eventIndex;
-      var location = traceEvents[index].location.first_line;
+      var line = traceEvents[index].location.first_line;
+      var appear_location = traceEvents[index].location;
+      var tracedLine = -1; 
+
+      //trace lines that are not animation.
+      while (currentRecordID < debugId) {
+        if (prevIndex != -1) {
+          var prevLocation = traceEvents[prevIndex].location;
+          var prevLine = prevLocation.first_line;
+        }
+        else {
+          var prevLocation = null;
+          var prevLine = -1;
+        }
+
+        if (tracedLine != -1) {
+          untraceLine(tracedLine);
+          tracedLine = -1;
+        }
+
+        if(currentLine < prevLine) {
+
+          if (arrows[prevIndex] != null) {
+            arrows[prevIndex]['after'] =  {first: currentLocation, second: prevLocation};
+          }
+          else {
+            arrows[prevIndex] = {before: null, after: {first: currentLocation, second: prevLocation}};
+          }
+          if (arrows[currentIndex] != null) {
+            arrows[currentIndex]['before'] =  {first: currentLocation, second: prevLocation};
+          }
+          else{
+            arrows[currentIndex] = {before: {first: currentLocation, second: prevLocation}, after : null};
+          }
+          view.arrow(view.paneid('left'), arrows, currentIndex, false);//should I pass in prevIndex and currentRecordID or?
+          debugRecordsByLineNo[prevLine].eventIndex = prevIndex;
+          debugRecordsByLineNo[currentLine].eventIndex = currentIndex;
+        }
+        traceLine(currentLine);
+        tracedLine = currentLine;
+        prevLine = currentLine;
+        prevIndex = debugRecordsByDebugId[currentRecordID].eventIndex;
```

Comment:
No. Index is the line being passed in by animation and prevIndex is whatever we previously traced. So, it can be equal to index-1 but may also be an earlier index that got traced by backtracking.